hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/cluster/ClusteringPlanActionExecutor.java

```diff
@@ -48,21 +48,28 @@
   private final Option<Map<String, String>> extraMetadata;
 
   public ClusteringPlanActionExecutor(HoodieEngineContext context,
-                                          HoodieWriteConfig config,
-                                          HoodieTable<T, I, K, O> table,
-                                          String instantTime,
-                                          Option<Map<String, String>> extraMetadata) {
+                                      HoodieWriteConfig config,
+                                      HoodieTable<T, I, K, O> table,
+                                      String instantTime,
+                                      Option<Map<String, String>> extraMetadata) {
     super(context, config, table, instantTime);
     this.extraMetadata = extraMetadata;
   }
 
   protected Option<HoodieClusteringPlan> createClusteringPlan() {
     LOG.info("Checking if clustering needs to be run on " + config.getBasePath());
-    Option<HoodieInstant> lastClusteringInstant = table.getActiveTimeline().getCompletedReplaceTimeline().lastInstant();
+    Option<HoodieInstant> lastClusteringInstant;
+    Option<HoodieInstant> pendingInstant = table.getActiveTimeline().filterPendingReplaceTimeline().lastInstant();
+    if (pendingInstant.isPresent()) {
```

Comment:
```java
table.getActiveTimeline().getInstants.()stream().filter(
        s -> s.getAction().equals(HoodieTimeline.REPLACE_COMMIT_ACTION)
```

guess we can filter the instants directly by the action type here ?

Reply:
> ```java
> table.getActiveTimeline().getInstants.()stream().filter(
>         s -> s.getAction().equals(HoodieTimeline.REPLACE_COMMIT_ACTION)
> ```
> 
> guess we can filter the instants directly by the action type here ?

@danny0405 good idea,I'm working on it.